docs/deposit_and_withdrawal.md

```diff
@@ -0,0 +1,102 @@
+# Deposit and withdrawal
+
+Deposit and withdrawal is a special layer1 <-> layer2 messaging mechanism with assets transfer.
+
+## Deposit
+
+User deposit by sending a layer1 transaction on CKB while generating a cell with a special lock - deposit lock, which allows the block producer to unlock the cell. The block producer unlocks the deposited cell and then moves the assets under the custodian lock, at the same time, the block producer writes the deposit assets to layer2 state. All the process is enforced by layer1 script, even the block producer can't take the assets away.
```

Comment:
```suggestion
A deposit request is a layer1 transaction which generates a cell with a special lock - deposit lock. The block producer will collect these cells and process the deposit requests in blocks. It will unlock the deposit cells, move the assets under the custodian lock, and update layer2 state, in block submission layer1 transactions. These transactions are checked by a layer1 script, so the block producer can't take the assets away.
```